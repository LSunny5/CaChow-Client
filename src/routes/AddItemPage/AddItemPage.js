import React, { Component } from 'react';
import '../../components/MenuItemForm/MenuItemForm.css';
import MenuItemForm from '../../components/MenuItemForm/MenuItemForm';
import PropTypes from 'prop-types';

export default class AddItemPage extends Component {
    render() {
        const { r_id } = this.props.match.params;
        return (
            <section className='RFormContent'>
                <h2 className="formTitle">Add Menu Items</h2>
                    <MenuItemForm restId={r_id}/>


                {/*//TODO:  do a submission after entire process is over*/}
               {/*  <form className="addForm" onSubmit={this.handleSubmit}> 
               <button onClick={this.addItem} className="addItemButton"> + Add item </button> 
               <div className="buttonBox extraMargin">
                        <button type='submit' className="button">
                            Add Item
                        </button> 
                         <NavLink
                            to={`/account`}
                            className="button"
                            onClick={this.alertCompleteUser}    >
                            Complete
                        </NavLink>
                        <NavLink
                            onClick={this.alertCancelUser}
                            className="button"
                            to={`/account`} >
                            Cancel
                        </NavLink>
                    </div>
                 </form> */}
            </section >
        )
    };
};

AddItemPage.propTypes = {
    r_id: PropTypes.string, 
}