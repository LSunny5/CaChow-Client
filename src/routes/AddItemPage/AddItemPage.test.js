import React from 'react';
import ReactDOM from 'react-dom';
import AddItemPage from './AddItemPage';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`AddItemPage component`, () => {
    const props = {
        match: {
            params: {
                r_id: 1
            }
        },
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <AddItemPage {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the AddItemPage form', () => {
        const wrapper = shallow(<AddItemPage {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});
