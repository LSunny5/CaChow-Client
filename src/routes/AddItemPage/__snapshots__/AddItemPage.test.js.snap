// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemPage component renders the AddItemPage form 1`] = `
<section
  className="RFormContent"
>
  <h2
    className="formTitle"
  >
    Add Menu Items
  </h2>
  <MenuItemForm
    restId={1}
  />
</section>
`;
